service/src/main/java/greencity/service/ubs/UBSClientServiceImpl.java

```diff
@@ -1254,31 +1267,89 @@ public void deleteOrder(Long id) {
     }
 
     @Override
-    public FondyOrderResponse processOrderFondyClientForIF(OrderFondyClientDto dto) throws Exception {
-        Order order = orderRepository.findById(dto.getOrderId()).orElseThrow();
+    public FondyOrderResponse processOrderFondyClientForIF(OrderFondyClientDto dto, String uuid) throws Exception {
+        Order order = orderRepository.findById(dto.getOrderId())
+            .orElseThrow(() -> new OrderNotFoundException(ORDER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+
+        User currentUser = userRepository.findUserByUuid(uuid)
+            .orElseThrow(() -> new UserNotFoundException(USER_WITH_CURRENT_ID_DOES_NOT_EXIST));
+
+        Map<Integer, Integer> amountOfBagsOrderedMap = order.getAmountOfBagsOrdered();
+
+        int sumToPay = formBagsToBeSavedAndCalculateOrderSumClient(amountOfBagsOrderedMap);
+
         if (order.getCounterOrderPaymentId() == null) {
             order.setCounterOrderPaymentId(0L);
         }
-        Order increment = incrementCounter(order);
-        PaymentRequestDto paymentRequestDto = formPaymentForIF(increment.getId(), dto.getSum());
-        Document doc = Jsoup.parse(restClient.getDataFromFondy(paymentRequestDto));
-        Elements links = doc.select("a[href]");
-        String link = links.attr("href");
-        return getPaymentRequestDto(order, link);
+
+        checkIfUserHaveEnoughPoints(currentUser.getCurrentPoints(), dto.getPointsToUse());
+        sumToPay = reduceOrderSumDueToUsedPoints(sumToPay, dto.getPointsToUse());
+
+        Set<Certificate> orderCertificates = new HashSet<>();
+        sumToPay = formCertificatesToBeSavedAndCalculateOrderSumClient(dto, orderCertificates, order, sumToPay);
+
+        currentUser.setCurrentPoints(currentUser.getCurrentPoints() - dto.getPointsToUse());
+        userRepository.save(currentUser);
+
+        if (sumToPay <= 0) {
+            order.setOrderPaymentStatus(OrderPaymentStatus.PAID);
+        }
+        order.setOrderStatus(OrderStatus.CONFIRMED);
+        orderRepository.save(order);
+        eventService.save(OrderHistory.ORDER_CONFIRMED, OrderHistory.SYSTEM, order);
+        if (sumToPay == 0) {
+            return getPaymentRequestDto(order, null);
+        } else {
+            Order increment = incrementCounter(order);
+            PaymentRequestDto paymentRequestDto = formPaymentForIF(increment.getId(), sumToPay);
+            Document doc = Jsoup.parse(restClient.getDataFromFondy(paymentRequestDto));
+            Elements links = doc.select("a[href]");
```

Comment:
introduce FoundyClientFacade class and incapsulate logic working with "a[href]" and other low-level details there. UBSClientServiceImpl doesn't need to work with such low-level details